### PR TITLE
(3/3) PaymentMethodList Tests: Refactor PaymentMethodList tests for improved readability

### PR DESCRIPTION
**Pull request by nauaros**
## Summary
Improves test readability and maintainability through helper functions, better assertions, and code organization.

## Changes
- Add `assertState()` and `assertApplePayButtonState()` helper functions in ViewModelTests
- Add test for `formattedAmount` with nil amount
- Add test for `onCancel` callback in action presenter
- Add MARK groupings to RouterTests for better organization
- Extract `StoredComponentMock` to shared mocks location
- Replace weak `#expect(true)` assertions with meaningful UI verification

## Testing
All tests pass.

## Dependencies
Depends on #2560

## Ticket [Optional]
<ticket>
COSDK-827
</ticket>

## Checklist

- [x] Tested changes locally
- [x] Added/updated unit tests
- [x] Verified against acceptance criteria

### CODE REVIEW

File: Tests/IntegrationTests/DropIn Tests/PaymentMethodList/PaymentMethodListViewControllerTests.swift [hunk 1]
```diff
@@ -139,8 +142,9 @@ struct PaymentMethodListViewControllerTests {
         viewModelMock.setState(.idle)
         await Task.yield()
 
-        // Then - stopLoading was called (no crash, state is idle)
-        #expect(true) // If we reach here, stopLoading didn't crash
+        // Then - loading overlay should be hidden (alpha == 0)
+        let loadingOverlay = sut.view.subviews.first { $0.backgroundColor?.cgColor.alpha ?? 1 < 1 }
```
[the remark this answers is not in the document]
Comment (nauaros) [reply Tests/IntegrationTests/DropIn Tests/PaymentMethodList/PaymentMethodListViewControllerTests.swift]: Changed

File: Tests/IntegrationTests/DropIn Tests/PaymentMethodList/PaymentMethodListViewControllerTests.swift [hunk 2]
```diff
@@ -115,12 +115,15 @@ struct PaymentMethodListViewControllerTests {
         // Allow state to propagate on main queue
         await Task.yield()
 
-        // Then - verify state was set (UI updates are handled internally)
-        #expect(true)
+        // Then - verify section views are added to the stack
+        let scrollView = sut.view.subviews.first { $0 is UIScrollView } as? UIScrollView
+        let contentStackView = scrollView?.subviews.first { $0 is UIStackView } as? UIStackView
+        let paymentMethodSectionsStackView = contentStackView?.arrangedSubviews.last { $0 is UIStackView } as? UIStackView
```
[the remark this answers is not in the document]
Comment (nauaros) [reply Tests/IntegrationTests/DropIn Tests/PaymentMethodList/PaymentMethodListViewControllerTests.swift]: Changed

File: Tests/IntegrationTests/DropIn Tests/PaymentMethodList/PaymentMethodListViewModelTests.swift [hunk 3]
```diff
@@ -16,6 +16,41 @@ import UIKit
...
+        isHidden: Bool = false,
+        isVisible: Bool = false
+    ) {
+        switch state {
+        case .hidden:
+            #expect(isHidden, "Expected applePayButtonState to be .hidden but got \(state)")
+        case .visible:
+            #expect(isVisible, "Expected applePayButtonState to be .visible but got \(state)")
+        }
+    }
```
[the remark this answers is not in the document]
Comment (nauaros) [reply Tests/IntegrationTests/DropIn Tests/PaymentMethodList/PaymentMethodListViewModelTests.swift]: Good one, I changed it

File: AdyenCheckout/CheckoutConfiguration/CheckoutConfiguration.swift [hunk 4]
```diff
@@ -38,14 +38,6 @@ public struct CheckoutConfiguration {
     // TODO: how we store configurations may change
     package var configurations: [CheckoutComponentType: CheckoutComponentConfiguration]
     
-    package var onSubmit: SubmitHandler?
```
Comment (robertdalmeida) [inline AdyenCheckout/CheckoutConfiguration/CheckoutConfiguration.swift]: nonblocking;
is this intentional. I think this is already removed from develop. maybe the brain points to an older develop.

File: Tests/IntegrationTests/DropIn Tests/PaymentMethodList/PaymentMethodListViewModelTests.swift [hunk 5]
```diff
@@ -126,11 +155,7 @@ struct PaymentMethodListViewModelTests {
         sut.didFail(with: error, from: paymentComponentMock)
 
         // Then
-        if case .idle = sut.state {
-            // Success
-        } else {
-            Issue.record("Expected state to be .idle after failure")
-        }
+        assertState(sut.state, isIdle: true)
```
Comment (robertdalmeida) [inline Tests/IntegrationTests/DropIn Tests/PaymentMethodList/PaymentMethodListViewModelTests.swift]: alternative approach; minor;
if we make the PaymentMethodListState equatable then 
we could write expect(sut.state == .idle)

If we add a new paymentmethodliststate then we would need to add another boolean to this helper method.

**Review by robertdalmeida — APPROVED** [review]

File: Tests/IntegrationTests/DropIn Tests/PaymentMethodList/PaymentMethodListViewModelTests.swift [hunk 6]
```diff
@@ -16,6 +16,43 @@ import UIKit
...
+    // MARK: - State Assertion Helper
+
+    private func assertState(
+        _ state: PaymentMethodListState,
+        isIdle: Bool = false,
+        isLoading: Bool = false,
+        isLoaded: Bool = false,
+        sectionCount: Int? = nil
+    ) {
+        let expectedState: String = isIdle ? ".idle" : isLoading ? ".loading" : isLoaded ? ".loaded" : "unknown"
```
Comment (atmamont) [inline Tests/IntegrationTests/DropIn Tests/PaymentMethodList/PaymentMethodListViewModelTests.swift]: This is a very confusing string conversion that could be done by a private extension for `PaymentMethodListState`, right? Then the assertion becomes a one-liner without the need to unwrap all cases.

File: Tests/IntegrationTests/DropIn Tests/PaymentMethodList/PaymentMethodListViewModelTests.swift [hunk 7]
```diff
@@ -16,6 +16,43 @@ import UIKit
...
+            }
+        }
+    }
+
+    private func assertApplePayButtonState(
+        _ state: PaymentMethodListHeaderViewModel.ApplePayButtonState,
+        isHidden: Bool = false,
+        isVisible: Bool = false
+    ) {
+        let expectedState: String = isHidden ? ".hidden" : isVisible ? ".visible" : "unknown"
```
Comment (atmamont) [inline Tests/IntegrationTests/DropIn Tests/PaymentMethodList/PaymentMethodListViewModelTests.swift]: The same here, let's try to work with data types not strings?

File: AdyenCheckout/CheckoutConfiguration/CheckoutConfiguration.swift (hunk 4 above)
Comment (nauaros) [reply AdyenCheckout/CheckoutConfiguration/CheckoutConfiguration.swift]: This is gone after syncing with develop

File: Tests/IntegrationTests/DropIn Tests/PaymentMethodList/PaymentMethodListViewModelTests.swift (hunk 5 above)
Comment (nauaros) [reply Tests/IntegrationTests/DropIn Tests/PaymentMethodList/PaymentMethodListViewModelTests.swift]: Much better to put some effort into making PaymentMethodListState Equatable. It is cleaner now

File: Tests/IntegrationTests/DropIn Tests/PaymentMethodList/PaymentMethodListViewModelTests.swift (hunk 6 above)
Comment (nauaros) [reply Tests/IntegrationTests/DropIn Tests/PaymentMethodList/PaymentMethodListViewModelTests.swift]: This is gone after implementing @robertdalmeida suggestion of making `PaymentMethodListState` Equatable

**Review by atmamont — APPROVED** [review]
